src/NoteWorkspace.ts

```diff
@@ -383,6 +394,84 @@ export class NoteWorkspace {
     );
   }
 
+  static newNoteFromSelection(context: vscode.ExtensionContext) {
+    const originEditor = vscode.window.activeTextEditor;
+
+    if (!originEditor) {
+      // console.debug('Abort: no active editor');
+      return;
+    }
+    const { selection } = originEditor;
+    const noteContents = originEditor.document.getText(selection);
+    const originSelectionRange = new vscode.Range(selection.start, selection.end);
+
+    if (noteContents === '') {
+      vscode.window.showErrorMessage('Error creating note from selection: selection is empty.');
+      return;
+    }
+    
+    // console.debug('newNote');
+    const inputBoxPromise = NoteWorkspace.showNewNoteInputBox();
+
+    inputBoxPromise.then(
+      (noteName) => {
+        if (noteName == null || !noteName || noteName.replace(/\s+/g, '') == '') {
+          // console.debug('Abort: noteName was empty.');
+          return false;
+        }
+        const { filepath, fileAlreadyExists } = NoteWorkspace.createNewNoteFile(noteName);
+        const destinationUri = vscode.Uri.file(filepath);
+
+        // open the file:
+        vscode.window
+          .showTextDocument(destinationUri, {
+            preserveFocus: false,
+            preview: false,
+          })
+          .then(() => {
+            if (!fileAlreadyExists) {
```

Comment:
If the file *does* already exist, neither the cut nor paste happens? Might be worth commenting the code as it is a fairly large `if` block we are aborting.